wiki-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java

```diff
@@ -1520,4 +1506,28 @@ private void initEmotionIcons() {
     }
   }
 
+  private void initTemplates() {
+    // init templates
+    for (WikiTemplatePagePlugin templatePlugin : templatePagePlugins_) {
+      if (templatePlugin != null && templatePlugin.getTemplates() != null) {
+        String wikiType = "";
+        String wikiOwner = "";
+        for (Template template : templatePlugin.getTemplates()) {
+          try {
+            wikiType = template.getWikiType();
+            wikiOwner = template.getWikiOwner();
+            Wiki wiki = new Wiki(wikiType, wikiOwner);
+            Wiki createdWiki = dataStorage.createWiki(wiki);

```

Comment:
Creating wikis in the method which creates the template does not seem very good in term of design. Templates should be created when the wikis are created, not the opposite.